File: test/fixtures/small-objc/src/Shape.m
#import "Shape.h"

@implementation Circle {
  double _radius;
}

- (instancetype)initWithRadius:(double)radius {
  self = [super init];
  if (self) {
    _radius = radius;
  }
  return self;
}

- (double)area {
  return 3.14159 * _radius * _radius;
}

@end

File: test/fixtures/small-cpp/src/shape.cpp
#include "shape.h"

namespace sample {

double Shape::describe() const {
  // Same-file out-of-line self-call: describe() -> area() (virtual dispatch).
  return area();
}

Circle::Circle(double radius) : radius_(radius) {}

double Circle::area() const {
  return kPi * radius_ * radius_;
}

}  // namespace sample
